xml/cl.xml

```diff
@@ -5864,15 +5884,17 @@ server's OpenCL/api-docs repository.
                 <type name="cl_image_pitch_info_qcom"/>
                 <type name="cl_mem_ext_host_ptr"/>
             </require>
-            <require comment="cl_mem_flags">
+            <require group="cl_mem_flags" etype="bitfield">
                 <enum name="CL_MEM_EXT_HOST_PTR_QCOM"/>
             </require>
-            <require comment="cl_device_info">
+            <require group="cl_device_info">
                 <enum name="CL_DEVICE_EXT_MEM_PADDING_IN_BYTES_QCOM"/>
                 <enum name="CL_DEVICE_PAGE_SIZE_QCOM"/>
             </require>
-            <require comment="cl_image_pitch_info_qcom">
+            <require group="cl_image_pitch_info_qcom">
+                <enum name="CL_IMAGE_ROW_PITCH"/>
                 <enum name="CL_IMAGE_ROW_ALIGNMENT_QCOM"/>
+                <enum name="CL_IMAGE_SLICE_PITCH"/>
```

Comment:
This isn't correct either.  `CL_IMAGE_ROW_PITCH` and `CL_IMAGE_SLICE_PITCH` are core enums are are not added by this extension.

```suggestion
            <require group="cl_image_pitch_info_qcom">
                <enum name="CL_IMAGE_ROW_ALIGNMENT_QCOM"/>
```

Reply:
Hmm, but these are newly added to `cl_image_pitch_info_qcom` otherwise:
```
Table 5.XXX

    List of supported param_names by clGetDeviceImageInfoQCOM

        cl_image_pitch_info_qcom       Return Type   Info returned in
                                                     param_value

        CL_IMAGE_ROW_PITCH             cl_uint       Returns the image row pitch
                                                     supported by this device

        CL_IMAGE_ROW_ALIGNMENT_QCOM    cl_uint       Returns the image row pitch
                                                     alignment supported by this
                                                     device

        CL_IMAGE_SLICE_PITCH           cl_uint       Returns the image slice
                                                     pitch supported by this
                                                     device

        CL_IMAGE_SLICE_ALIGNMENT_QCOM  cl_uint       Returns the image slice
                                                     pitch alignment supported
                                                     by this device
```
I think I already saw multiple extensions adding the same enums - in case either one of them is supported. Is it actually wrong for extension to add relevant enums again, even if they are in core?

Reply:
I think we may have a different idea what should be included in the XML file.  To be honest, I'm not sure if we have this written down anywhere (assuming my understanding is correct); maybe @oddhack can help.

My understanding is that the XML file is describing the types, enums, APIs, etc. that are *added* by a new extension, on top of whatever is supported in the core API.  If two extensions happen to add the same new enum (or type, or API), and neither extension has a dependency on the other extension, that's OK and both extensions can require the same new enum, because extensions are intended to stand on their own.  An extension would never require a core API enum explicitly though - instead, this requirement would be indicated by requiring a specific API version.

In the case above, `cl_image_pitch_info_qcom` can accept the `CL_IMAGE_ROW_PITCH` and `CL_IMAGE_SLICE_PITCH` enums, but these enums already exist in the core API and are not added by the extension.  Therefore, I don't think they should be required by the extension, just like e.g. the `cl_uint` type is not required by the extension.  Does this make sense?

Reply:
Well, deciding if something should exist in XML without the context is indeed arbitrary.
But in this case, the reason for this info to exist in XML is the same as the reason for this PR: The consistent grouping of enums.

The extension adds functionality to these core enums (attached `cl_uint` return type, which I'm planning to cover in a future PR), and for that, also adds them to the `cl_image_pitch_info_qcom` enum group (which this PR's purpose covers). I don't know a better way to describe the fact that this extension adds 2 core enums to its new group.

Meanwhile, my argument about other extensions adding the same enum is more about the fact that we cannot break anything this way - since any code should already be capable of handling this.
Actually, a better example should be the `cl_mem_flags` and `cl_svm_mem_flags` groups, which add some of the same enums in the 1.0 version of the core in this PR.